src/main/java/org/stocker/routes/UpdateRecordsRoute.java

```diff
@@ -41,6 +45,16 @@ public UpdateRecordsRoute(StockDataClient stockDataClient, StockDBClient stockDB
     @Override
     public void handle(YokeRequest yokeRequest, Handler<Object> handler) {
         YokeResponse response = yokeRequest.response();
+        StockMarket market = null;
+        if (yokeRequest.getParameter("market").equalsIgnoreCase(StockMarket.INDIA.toString())) {
```

Comment:
There is no need to have this logic explicitly . We can build a middleware which automatically handles this for all endpoints requiring a ```market``` . Else we would end up writing this ```if else ``` block pretty much in every single handler .